File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q31-count-the-repetitions.js
/**
 * @param {string} s1
 * @param {number} n1
 * @param {string} s2
 * @param {number} n2
 * @return {number}
 */
var getMaxRepetitions = function(s1, n1, s2, n2) {
    const map = new Map();
    let count = 0;
    let i = 0;
    let j = 0;
    while (count < n1) {
        if (j === s2.length) {
            count++;
            j = 0;
        }
        if (map.has(i)) {
            const lastIndex = map.get(i);
            const repeatCount = (i - lastIndex) / (j - map.get(lastIndex));
            const remaining = (n1 - count) % repeatCount;
            if (remaining === 0) {
                return (n1 - count) / repeatCount * n2;
            }
            i = lastIndex + repeatCount * (j - map.get(lastIndex));
        } else {
            map.set(i, j);
            i += s1.indexOf(s2[j]);
            j++;
        }
    }
    return count * n2;
};

File: ELEC881_Copilot/src/Function Name Masked/Python/q31-count-the-repetitions.py
class ab(object):
    def get(self, s1, n1, s2, n2):
        if not s1 or not s2 or n1 == 0 or n2 == 0:
            return 0
        s1 = list(s1)
        s2 = list(s2)
        i = 0
        j = 0
        count = 0
        while count < n1:
            if s1[i] == s2[j]:
                j += 1
                if j == len(s2):
                    j = 0
                    count += 1
            i += 1
            if i == len(s1):
                i = 0
        return count